src/debug/debugSession.ts

```diff
@@ -104,6 +105,89 @@ export class DebugSession implements IDebugSession {
         });
     }
 
+    /**
+     * In attach mode, the user should choose the running pod first, then the debugger will attach to it.
+     *
+     * @param workspaceFolder the active workspace folder.
+     * @param pod the target pod name.
+     */
+    public async attach(workspaceFolder: vscode.WorkspaceFolder, pod?: string): Promise<void> {
+        if (!workspaceFolder) {
+            return;
+        }
+
+        // Select the image type to attach.
+        this.debugProvider = await providerRegistry.getDebugProvider();
+        if (!this.debugProvider) {
+            return;
+        } else if (!await this.debugProvider.isDebuggerInstalled()) {
+            return;
+        }
+
+        // Select the target pod to attach.
+        let targetPod = pod, targetContainer, containers = [];
+        const resource = await kubectlUtils.getResourceAsJson(this.kubectl, pod ? `pod/${pod}` : "pods");
+        if (!resource) {
+            return;
+        }
+        if (pod) {
+            containers = resource.spec.containers;
+        } else {
+            const podPickItems: vscode.QuickPickItem[] = resource.items.map((pod) => {
+                return {
+                    label: `${pod.metadata.name} (${pod.spec.nodeName})`,
+                    description: "pod",
+                    name: pod.metadata.name,
+                    containers: pod.spec.containers
+                };
+            });
+            const selectedPod = await vscode.window.showQuickPick(podPickItems, { placeHolder: `Please select a pod to attach` });
+            if (!selectedPod) {
+                return;
+            }
+            targetPod = (<any> selectedPod).name;
+            containers = (<any> selectedPod).containers;
+        }
+
+        // Select the target container to attach.
+        if (containers.length > 1) {
+            const containerPickItems: vscode.QuickPickItem[] = containers.map((container) => {
+                return {
+                    label: `${container.name} (${container.image})`,
+                    description: "container",
+                    name: container.name
+                };
+            });
+            const selectedContainer = await vscode.window.showQuickPick(containerPickItems, { placeHolder: "Please select a container to attach" });

```

Comment:
Same comment to insert blank lines between code blocks. Suggest a following blank line after control blocks like if/while/for or function/lambda.